ons-spark/spark-analysis/big-data-workflow.md

```diff
@@ -0,0 +1,118 @@
+- Follow the guidance on the [optimising joins](../spark-concepts/join-concepts) page of this book.
+
+### 11. Consider data [partitioning](../spark-concepts/partitions)
+- Spark splits data into partitions to perform parallel processing. 
+- Having too many or too few partitions may cause performance issues in your code. 
+- There is no set rule for how many partitions your data should have. As a general guide, the optimum size for a partition is often suggested to be 100-200MB and the number of partitions should be a multiple of the number of nodes in your [Spark session](../spark-overview/spark-session-guidance). 
+- It may be beneficial to change how your data is partitioned at various points in your pipeline. For example if your full dataset contains 200 million rows, it may be reasonable to have 200 partitions. However, if later on in your pipeline this data is aggregated so that there are now only around 2 million rows, repartitioning your data on to a smaller number of partitions could improve performance and reduce the compute resource used. 
+- Aim to make sure data is kept in roughly equal partition sizes throughout the pipeline. 
+    - This may also involve repartitioning your data at appropriate points due to [skewed data](../spark-concepts/partitions#intermediate-partitions-in-wide-operations)
+    - Identifying variables in your data that are skewed during the EDA phase of your pipeline development may help in avoiding or identifying this issue later in the development stage. 
+- Be aware of the difference between [*narrow* and *wide* transformations](../spark-concepts/shuffling) in Spark and how *wide* transformations can increase the number of partitions in your data. 
+    - If you are using Spark 2.X, you may need to repartition after performing such operations or consider changing the `spark.sql.shuffle.partitions` from the default (200) in your [Spark config](../spark-overview/spark-defaults).
+    - If you are using Spark 3.X, ensure [Adaptive Query Excution (AQE)](https://spark.apache.org/docs/latest/sql-performance-tuning.html#adaptive-query-execution) is switched on by setting `spark.sql.adaptive.enabled` to `true` in your [Spark config](../spark-overview/spark-defaults). This will automatically coalesce post-shuffle partitions.
+    
+### 12. Become familiar with the Spark UI    
+- The [Spark UI](../spark-concepts/spark-application-and-ui) is used to monitor the status and resource consumption of your Spark cluster and can be a useful tool for troubleshooting slow Spark code.
+- Check the size of the execution plan regularly
+    - Do this either by examining the DAG (Directed Acyclic Graph) in the Spark UI or by using `df.explain()`/`explain(df)` in PySpark/SparklyR.
+    - Overly long execution plans can cause performance issues.
+    - Consider using [checkpoints and staging tables](../spark-concepts/checkpoint-staging) at appropriate points in your code to break the lineage in these cases.
+    
```

Comment:
This isn't essential - but I think having a short closing paragraph just to bring everything together might be a good idea because it feels like it ends quite abruptly.

Reply:
I think that's a good idea. I've attempted to add something to sum it all up :)